modules/reactor/include/meshgenerators/PinMeshGenerator.h

```diff
@@ -32,49 +34,68 @@ class PinMeshGenerator : public ReactorGeometryMeshBuilderBase
   // Define metadata associated with PinMeshGenerator
   void generateMetadata();
 
-  ///The id number for this pin type
+  /**
+   * Get CSGSurfaces corresponding to hexagonal or square region with given halfpitch
+   * @param radial_index Radial index of hex / square region, for surface naming
+   * @param halfpitch Halfpitch of square or hexagon
+   * @param csg_obj Reference to CSGBase object for adding defined surfaces to
+   * @return vector of surfaces that correspond to hexagonal or square region
+   */
+  std::vector<std::reference_wrapper<const CSG::CSGSurface>> getHexCartSurfaces(
+      unsigned int radial_index, Real halfpitch, std::unique_ptr<CSG::CSGBase> & csg_obj);
```

Comment:
```suggestion
      unsigned int radial_index, Real halfpitch, CSG::CSGBase & csg_obj);
```
can we do that? we should not be passing unique pointers around, the memory ownership would be expected to come with

Reply:
You're right, updated